scripts/update-index.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# This script update 'docs/index.md' with 'README.md' 
+#  changing github/learn-ocaml/docs URLs to relative path.
+
+readme="$PWD/README.md"
+index="$PWD/docs/index.md"
+
+if [[ ! -r "$readme" ]]
+then
+  echo "Error: '$readme': file does not exist or unreadable"
+  exit 1 
+fi
+
+if [[ ! -w "$index" ]]
+then
+  echo "Error: '$index': file does not exist or unwritable"
+  exit 1 
+fi
+
+buffin=$(<$readme)
+buffout=""
+
+url="https://github.com/ocaml-sf/learn-ocaml/blob/master/docs"
+
+while IFS= read -r line
+do
+
+  # Using sed command with "; semicolon" separator to avoid any problems 
+  #  with the "/ slash" contained in URLs.
+  line=$(echo "$line" | sed -r -e 's;'"$url"';.;g')
```

Comment:
Always avoid the pattern `echo "$line"`
(because if `line` = `-e` or `-n`, it is eaten…)

Instead, write:
* `printf '%s\n' "$line"`
* or `printf '%s' "$line"`

Reply:
I see what you mean for the comsuption of `-e` and `-n`, I will change the approach.